### PR TITLE
Use ElementTree to validate xml

### PR DESCRIPTION
**Pull request by proteusvacuum**
code buddy @millerdev 
Follow up from: https://github.com/dimagi/commcare-hq/pull/12517#issuecomment-233554434

### CODE REVIEW

File: corehq/apps/fixtures/tests.py [hunk 1]
```diff
@@ -338,6 +339,14 @@ def test_combo(self):
         bad_name = "<space>\<dadgg sd"
         self.assertTrue(is_identifier_invalid(bad_name))
 
+    def test_starts_with_number(self):
+        bad_name = "0hello"
+        self.assertTrue(is_identifier_invalid(bad_name))
+
+    def test_unicode(self):
+        bad_name = u"ﾉｲ丂 ﾑ ｲ尺ﾑｱ! \_(ツ)_/¯"
+        self.assertTrue(is_identifier_invalid(bad_name))
```
Comment (millerdev) [inline corehq/apps/fixtures/tests.py]: completely not related to this PR: I suppose the original use case for this function made more sense with a negative in the function name, but it takes a moment to adjust. When I'm speed reading I see "is valid identifier" (the opposite).

File: corehq/apps/fixtures/utils.py [hunk 2]
```diff
@@ -22,9 +23,12 @@ def is_identifier_invalid(name):
     - Start with "xml"
     - Start with a number
     """
-    return (
-        not bool(name) or
-        bool(re.search(BAD_SLUG_PATTERN, name)) or
-        name.startswith('xml') or
-        name[0].isdigit()
-    )
+    if (not bool(name) or name.startswith('xml')):
```
Comment (millerdev) [inline corehq/apps/fixtures/utils.py]: nit: `bool(name)` is the same as `name` in this context

**Comment by millerdev** [comment]
👍 on ✅

File: corehq/apps/fixtures/tests.py (hunk 1 above)
Comment (proteusvacuum) [inline corehq/apps/fixtures/tests.py]: Completely agree. I actually first wrote it to return `False` when it was bad, but then all the tests failed...

File: corehq/apps/fixtures/tests.py [hunk 3]
```diff
@@ -338,6 +339,14 @@ def test_combo(self):
         bad_name = "<space>\<dadgg sd"
         self.assertTrue(is_identifier_invalid(bad_name))
 
+    def test_starts_with_number(self):
+        bad_name = "0hello"
+        self.assertTrue(is_identifier_invalid(bad_name))
+
+    def test_unicode(self):
+        bad_name = u"ﾉｲ丂 ﾑ ｲ尺ﾑｱ! \_(ツ)_/¯"
```
Comment (czue) [inline corehq/apps/fixtures/tests.py]: :laughing:

**Pull request merged**